traffic_ops/traffic_ops_golang/profile/profiles.go

```diff
@@ -269,13 +271,13 @@ JOIN profile_parameter pp ON pp.parameter = p.id
 WHERE pp.profile = :profile_id`
 }
 
-func (pr *TOProfile) checkIfProfileCanBeAlteredByCurrentUser() (error, error, int) {
+func canProfileBeAlteredByCurrentUser(user string, tx *sql.Tx, cName *string, cdnID *int) (error, error, int) {
 	var cdnName string
-	if pr.CDNName != nil {
-		cdnName = *pr.CDNName
+	if cName != nil {
+		cdnName = *cName
 	} else {
-		if pr.CDNID != nil {
-			cdn, ok, err := dbhelpers.GetCDNNameFromID(pr.ReqInfo.Tx.Tx, int64(*pr.CDNID))
+		if cdnID != nil {
+			cdn, ok, err := dbhelpers.GetCDNNameFromID(tx, int64(*cdnID))
```

Comment:
can we instead just not allow these parameters to be passed as `nil` by making them non-reference values?

Reply:
We could.